daemon/daemon.go

```diff
@@ -647,7 +647,15 @@ func (daemon *Daemon) registerLink(parent, child *container.Container, alias str
 
 // DaemonJoinsCluster informs the daemon has joined the cluster and provides
 // the handler to query the cluster component
-func (daemon *Daemon) DaemonJoinsCluster(clusterProvider cluster.Provider) {
+func (daemon *Daemon) DaemonJoinsCluster(clusterProvider cluster.Provider, forceNewCluster bool) {
+	// When forcing a new cluster, first clean up the existing agent
+	// ensuring that a new one will be created and started
+	if(forceNewCluster) {
+		daemon.setClusterProvider(nil)
```

Comment:
I still see this as not super clean. If the Agent is stopped, I believe libnetwork need to nullify the previous cluster provider automatically. Is there any other case where the clusterProvider is being reused without being set?

Reply:
You're right there's another way that might be better. I've updated https://github.com/docker/libnetwork/pull/2307. Now instead the cluster provider would be set to nil in agentClose.